drgadget/ropviewer.py

```diff
@@ -295,18 +320,21 @@ def update_content_viewers(self, n=None):
                     perms += "W" if perm & idaapi.SEGPERM_WRITE != 0 else "."
                     perms += "X" if perm & idaapi.SEGPERM_EXEC != 0 else "."
                     self.iv.add_line("<%s> [%X - %X], %s, [%s]" % (name, ea_start, ea_end, ltype, perms))
-            else:
-                stype = GetStringType(item.ea)
-                if stype is not None:
-                    scontent = GetString(item.ea, -1, stype)
-                    if scontent != None and len(scontent):
-                        self.dav.add_line(idaapi.COLSTR("\"%s\"" % scontent, idaapi.SCOLOR_DSTR))
-                        # length = idaapi.get_max_ascii_length(item.ea, stype, idaapi.ALOPT_IGNHEADS)
-                        # self.hv.add_line()
-                else:
-                    scontent = GetString(item.ea, -1, ASCSTR_C)
-                    if scontent != None and len(scontent):
-                        self.dav.add_line("\"%s\"" % scontent)
+            elif item.type == Item.TYPE_ADDRESS:
+                # add a hex dump and a string view (if it's a string)
+                hex_dump_length = HEX_DUMP_DISPLAY_LENGTH
+
+                string_type = GetStringType(item.ea)
```

Comment:
Yes, addresses pointing into substrings is one example for what I meant by "not yet present or invalid". If I recall, this is what someone added the comment function for, at some point. We could either stick to that or add a feature that requires the user to manually specify a type of string. Alternatively or in addition, an automated way of detecting strings could involve simple heuristics (e.g. 4 " printable" characters at a minimum followed by a null byte or something - ASCII/Unicode?). There are at least two functions exported by the SDK for reading bytes - get_byte() and get_dbg_byte() for reading data from loader segments and debug segments, respectively, if I am not mistaken (currently no access to SDK). So I would say it's up to what you want to implement :)